### PR TITLE
New aos protocol

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: aosprotocol/aosprotocol.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#pragma pack(push, 1)
+
+/**
+ * Aos low level protocol header.
+ */
+struct AosProtocolHeader {
+    uint32_t mPort;
+    uint32_t mDataSize;
+    uint8_t mCheckSum[32];
+    uint8_t mData[];
```
Comment (MykolaSolyanko) [inline aosprotocol/aosprotocol.h]: variable length array is not part of standard C++, it possible to get compiler warning or error if we compile our code with pedantic flag

File: aosprotocol/aosprotocol.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
```
Comment (MykolaSolyanko) [inline aosprotocol/aosprotocol.h]: 2024

**Review by mlohvynenko — APPROVED** [review]

File: aosprotocol/aosprotocol.h (hunk 2 above)
Comment (MykolaSolyanko) [reply aosprotocol/aosprotocol.h]: still relevant

**Review by MykolaSolyanko — APPROVED** [review]

File: aosprotocol/aosprotocol.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Aos low level protocol header.
+ */
+struct AosProtocolHeader {
+    uint32_t mPort;
+    uint32_t mDataSize;
+    uint8_t mCheckSum[32];
+};
+
+struct AosProtobufHeader {
```
Comment (mykola-kobets-epam) [inline aosprotocol/aosprotocol.h:25]: no struct description

File: proto/servicemanager/v4/servicemanager.proto [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    bytes       sha256       = 6;
+    bytes       sha512       = 7;
+    uint64      size         = 8;
+}
+
+message LayerInfo {
+    VersionInfo version_info = 1;
+    string      url          = 2;
+    string      layer_id     = 3;
+    string      digest       = 4;
```
Comment (mykola-kobets-epam) [inline proto/servicemanager/v4/servicemanager.proto]: shouldnt digest be byte array?

File: proto/servicemanager/v4/servicemanager.proto [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+message VersionInfo {
+    uint64 aos_version    = 1;
+    string vendor_version = 2;
+    string description    = 3;
+}
+
+message FirewallRule {
+    string dst_ip   = 1;
+    string dst_port = 2;
+    string proto    = 3;
```
Comment (mykola-kobets-epam) [inline proto/servicemanager/v4/servicemanager.proto:84]: src port probably?

File: proto/servicemanager/v4/servicemanager.proto [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+
+message OverrideEnvVarStatus {
+    repeated EnvVarInstanceStatus env_vars_status = 1;
+    common.v1.ErrorInfo           error           = 2;
+}
+
+message EnvVarInstanceStatus {
+    InstanceIdent         instance    = 1;
+    repeated EnvVarStatus vars_status = 2;
+}
```
Comment (mykola-kobets-epam) [inline proto/servicemanager/v4/servicemanager.proto:227]: EnvVarStatus is used before declaration

File: proto/servicemanager/v4/servicemanager.proto [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+message NodeMonitoring {
+    google.protobuf.Timestamp   timestamp           = 1;
+    MonitoringData              monitoring_data     = 2;
+    repeated InstanceMonitoring instance_monitoring = 3;
+}
+
+message MonitoringData {
+    uint64                  ram         = 1;
+    uint64                  cpu         = 2;
+    repeated PartitionUsage disk        = 3;
```
Comment (mykola-kobets-epam) [inline proto/servicemanager/v4/servicemanager.proto:257]: disks to be uniform with other repeated fields

File: aosprotocol/aosprotocol.h (hunk 1 above)
Comment (al1img) [reply aosprotocol/aosprotocol.h]: removed

File: aosprotocol/aosprotocol.h (hunk 2 above)
Comment (al1img) [reply aosprotocol/aosprotocol.h]: fixed

File: proto/common/v1/common.proto [hunk 8]
```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+
+package common.v1;
+
+option go_package = "github.com/aoscloud/aos_core_api/proto/common";
+
+message ErrorInfo {
+    int32 AosCode  = 1;
```
[the remark this answers is not in the document]
Comment (al1img) [reply proto/common/v1/common.proto:8]: done

File: proto/servicemanager/v3/servicemanager.proto [hunk 9]
```diff
@@ -319,8 +314,8 @@ message ResourceValidateAlert {
 }
 
 message ResourceValidateErrors {
-    string name               = 1;
-    repeated string error_msg = 2;
+    string name                            = 1;
+    repeated common.v1.ErrorInfo error_msg = 2;
```
[the remark this answers is not in the document]
Comment (al1img) [reply proto/servicemanager/v3/servicemanager.proto:318]: done

File: aosprotocol/aosprotocol.h (hunk 3 above)
Comment (al1img) [reply aosprotocol/aosprotocol.h:25]: done

File: proto/servicemanager/v4/servicemanager.proto (hunk 4 above)
Comment (al1img) [reply proto/servicemanager/v4/servicemanager.proto]: Digest is string for layer everywhere in Aos code. Don't want to change it.

File: proto/servicemanager/v4/servicemanager.proto (hunk 5 above)
Comment (al1img) [reply proto/servicemanager/v4/servicemanager.proto:84]: proto stands for protocol. Src port is not used.

File: proto/servicemanager/v4/servicemanager.proto (hunk 7 above)
Comment (al1img) [reply proto/servicemanager/v4/servicemanager.proto:257]: done

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**